dali/python/nvidia/dali/_autograph/utils/tf_stack.py

```diff
@@ -148,42 +170,54 @@ def get_filtered_filenames(self):
         self._cached_set = filtered_filenames
         return filtered_filenames
 
+    def is_filtered(self, filename):
+        return filename in self.internal_set
 
-def extract_stack(stacklevel=1):
-    """An eager-friendly alternative to traceback.extract_stack.
 
-    Args:
-      stacklevel: number of initial frames to skip when producing the stack.
+class CustomModuleFilter(StackTraceFilter):
+    """Filters stack frames from the modules that were listed for this filter.
 
-    Returns:
-      A list-like FrameSummary containing StackFrame-like objects, which are
-      namedtuple-like objects with the following fields: filename, lineno, name,
-      line, meant to masquerade as traceback.FrameSummary objects.
+    We detect the top directory of given module and filter all frames from that path or its subpath.
     """
-    thread_key = _get_thread_key()
-    return _tf_stack.extract_stack(
-        _source_mapper_stacks[thread_key][-1].internal_map,
-        _source_filter_stacks[thread_key][-1].internal_set,
-        stacklevel,
-    )
 
+    def __init__(self, module_filter: "Union[List[ModuleType], ModuleType]"):
+        super().__init__()
+        self._filtered_filenames = set()
+        if not isinstance(module_filter, list):
+            module_filter = [module_filter]
+        for module in module_filter:
+            try:
+                module_file = inspect.getfile(module)
+                init_py = "/__init__.py"
+                if module_file.endswith(init_py):
+                    module_file = module_file[: -len(init_py)]
```

Comment:
That's a bit hacky. How about just splitting on `/`, discarding last item if it's `__init__.py` and then joining?

Reply:
Tbh I don't see much of a difference in how we detect and remove the `__init__.py` that indicates a module directory.

Reply:
I know we're not windows-ready, but how does Python represent paths on windows? Shouldn't we look for directory separator (or replace it with `/` before this check)?

Reply:
I removed the `/` from the comparison string, the filtering should work with training directory separator as well.